### PR TITLE
[PyTorch] Use reduction=none for train_epoch_ch3

### PR DESCRIPTION
**Pull request by AnirudhDagar**
*Description of changes:*
Fixes loss not showing up in the following sections:
* https://d2l.ai/chapter_linear-networks/softmax-regression-concise.html
* https://d2l.ai/chapter_multilayer-perceptrons/mlp-scratch.html
* https://d2l.ai/chapter_multilayer-perceptrons/mlp-concise.html
* https://d2l.ai/chapter_multilayer-perceptrons/dropout.html

I'll look at other sections that might be affected. Note that `l.mean()` is required otherwise when using `l.sum()` with PyTorch CrossEntropy (with no reduction) the loss overshoots. This change will not affect any other notebooks because `d2l.train_ch3` is only used in these notebooks, so it won't break any other section.

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

**Comment by mli** [comment]
Job d2l-en/PR-2007/1 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-2007/
